plugin/server/services/chunks.js

```diff
@@ -40,70 +45,116 @@ function splitTextArray(textArray, { maxLength, maxByteSize } = {}) {
           'Splitting is currently achieved by splitting at one or multiple new lines and joining back together with one new line.'
       )
 
-      // FIXME: Splitting at multiple newlines might be breaking content layout
-      const splitTextField = textField.split(/\n+/)
+      let splitTextField
+      let joinChar = '\n'
+      let splitChar
 
-      // Recursively call this method to split the array into chunks of appropriate length and size
-      const { chunks } = splitTextArray(splitTextField, {
-        maxLength,
-        maxByteSize,
-      })
+      if (textField.includes('\n')) {
+        // FIXME: Splitting at multiple newlines might be breaking content layout
+        splitTextField = textField.split(/\n+/)
+      } else if (textField.includes('. ')) {
+        splitChar = '.'
+      } else if (textField.includes('? ')) {
+        splitChar = '?'
+      } else if (textField.includes('! ')) {
+        splitChar = '!'
```

Comment:
This will not work for all languages. Not sure how that could be handled. There probably exists an EOS (end of sentence) library somewhere.